tests/providers/amazon/aws/hooks/test_redshift_data.py

```diff
@@ -0,0 +1,76 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import unittest
+from unittest import mock
+
+from airflow.providers.amazon.aws.hooks.redshift_data import RedshiftDataHook
+
+
```

Comment:
For this test, check that we can be able to connect and that we are actually calling the API we connected to. The current tests doesn't guarantee that we are hitting that API

Reply:
Working on adding a system test and improving the unit test...will update PR when ready.  Thanks!

Reply:
FYI. I am taking over John over as per today. I am planning to improve unit tests and add system tests. I just realized [moto](https://github.com/spulec/moto) does not cover redshift-data. I am planning to add it in moto, and then I'll be able to move forward this PR

Reply:
this is a very simple hook / operator and i think using moto may be overcomplicating it. we can just write simple unit tests and conventional mocking techniques.  there are very few moving parts. basically we should just need to check that 1. execute_statement should be called with the operator params and 2. execution statuses should be handled properly, 3 nowait behavior, 4 xcom behavior. if moto is still holding you back i would not worry about it.